feat: :sparkles: introduce ip param for wrangler login command, fixes [5937]

gnekich opened the pull request:
fix issue #5937 by providing option to user to pass ip for the temporary login server, enabling the use of wrangler login command in devcontainers

Fixes #5937.

I would also argue that the default should be '0.0.0.0' for the values, as it is temporary login server, available for a short period. Making DX a lot better than to add --ip '0.0.0.0' every time in devcontainers, but for now PR defaults to localhost as that was the default before.

---

<!--
Please don't delete the checkboxes <3
The following selections do not need to be completed if this PR only contains changes to .md files
-->

- Tests
  - [ ] maybe for TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because: any regression would be captured by pre-existing tests in `__tests__/user.test.ts`
- Wrangler E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: covered by unit tests
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/21737
  - [ ] Documentation not necessary because:
- V3 wrangler backport: not necessary because new feature

petebacondarwin reviewed (changes requested):
Thanks for putting up this PR @gnekich.
I think we should use `host` rather than `ip` throughout (e.g. `localhost` is not an IP address).

I also think we should be more clear what these args are - host/ip and port are quite ambiguous and overloaded terms. When I first saw this PR I thought you were changing the host:port of our OAuth endpoint.
How about we use the terms `callback-host` and `callback-port`?

Finally, we will need a changeset (with a minor version bump) and a test.

gnekich commented:
Thanks for the feedback, @petebacondarwin! I've updated the PR and renamed the arguments to 'callback-host' and 'callback-port' for clarity. Also added a changeset with a minor version bump and a test.

petebacondarwin commented:
Thanks @gnekich - I'll follow up with the docs updates etc and merge this later this month after DevWeek.
